.github/workflows/check.yml

```diff
@@ -1,16 +1,45 @@
-name: ESLint and Prettier
+name: CI Checks
 
 on:
   push:
-    branches: [ master ]
     paths:
+        # rerun when typescript files change
       - "ts/**"
-  pull_request:
-    branches: [ master ]
+        # rerun when workflow is changed
+      - ".github/workflows/check.yml"
+        # rerun when packages change
+      - "package.json"
+      - "package-lock.json"
+  pull_request: {}
```

Comment:
```suggestion
  pull_request:
```

Is the empty dict needed?

Reply:
Probably not actually! I'll try it without!

Reply:
Yep seems to work fine!